python/AMI_NOT_PUBLIC_CHECK/AMI_NOT_PUBLIC_CHECK.py

```diff
@@ -0,0 +1,363 @@
+# Copyright 2017-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the License is located at
+#
+#        http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for
+# the specific language governing permissions and limitations under the License.
+
+'''
+#####################################
+##           Gherkin               ##
+#####################################
+Rule Name:
+  AMI_NOT_PUBLIC_CHECK
+
+Description:
+  Check whether the Amazon Machine Images are not publicly accessible. The rule is NON_COMPLIANT if one or more Amazon Machine Images are publicly accessible.
+
+Trigger:
+  Periodic
+
+Reports on:
+  AWS::::Account
+
+Rule Parameters:
+  None
+
+Scenarios:
+  Scenario: 1
+    Given: No AMIs with "is-public" parameter set to True
+     Then: Return COMPLIANT
+  Scenario: 2
+    Given: One or more AMIs with is-public parameter set to True
+     Then: Return NON_COMPLIANT with Annotation containing AMIIDs
+'''
+
+import json
+import sys
+import datetime
+import boto3
+import botocore
+
+try:
+    import liblogging
+except ImportError:
+    pass
+
+# Define the default resource to report to Config Rules
+DEFAULT_RESOURCE_TYPE = 'AWS::::Account'
+
+# Set to True to get the lambda to assume the Role attached on the Config Service (useful for cross-account).
+ASSUME_ROLE_MODE = False
+
+# Other parameters (no change needed)
+CONFIG_ROLE_TIMEOUT_SECONDS = 900
+
+# Generates list of image_id's of public images
+def generate_image_id_list(images, event):
+    image_ids = []
+    for image in images:
+        image_ids.append(image['ImageId'])
+    return image_ids
+
+def evaluate_compliance(event, configuration_item, valid_rule_parameters):
+    ec2_client = get_client('ec2', event)
+    public_ami_result = ec2_client.describe_images(
+                              Filters=[
+                                  {
+                                      'Name': 'is-public',
+                                      'Values': ['true']
+                                  }
+                              ],
+                              Owners=[event['accountId']]
+                        )
+    # If public_ami_list is not empty, generate non-compliant response
+    if public_ami_result['Images']:
+                image_ids = generate_image_id_list(public_ami_result['Images'], event)
+                return build_evaluation(
+                            event['accountId'],
+                            "NON_COMPLIANT",
+                            event,
+                            annotation="Amazon Machine Image Id: {} is public.".format(image_ids)[:256]
```

Comment:
It is a quick way to do, but it won't work well. Since you are going to cut "is public." first (or not clean cut on image_ids, not sure about the order of operations here). 

Please rephrase this way `Amazon Machine Image(s) publicly accessible: `. Then when you truncate, please add a ` (truncated)` at the end to make it clearly visible to the Customers.

Please make a test about it too.

Reply:
Changed the annotation.